lib/board.rb

```diff
@@ -69,13 +70,31 @@ def same_symbole?(array, mark)
   end
 
   def win_with_diagonal?(mark)
-    if board[0][0] == mark && board[1][1] == mark && board[2][2] == mark
-      true
-    elsif board[0][2] == mark && board[1][1] == mark && board[2][0] == mark
-      true
-    else
-      false
+    check_diag_one(mark) || check_diag_two(mark)
+  end
+
+  def check_diag_one(mark)
+    result = true
+
+    board.each_with_index do |value, index|
+      if board[index][index] != mark
+        result = false
+      end
+    end
+
+    result
+  end
+
+  def check_diag_two(mark)
+    result = true
+
+    board.each_with_index do |value, index|
+      if board[index][board.size - index - 1] != mark
```

Comment:
I finally used all?. I created a method to make it more readable, not really sure about the name at the moment "backward_index(index)"